cmd/internal/shell/core.go

```diff
@@ -49,13 +49,23 @@ func Get() Shell {
 //   `%F{cyan}wash ${prompt_path}%F{green} ❯%f `
 // with substitution for shell-specific portions of the function.
 func preparePrompt(cyan, green, reset, assign string) string {
+	// If PWD is a subdirectory of $W, then strip $W/. By removing the slash if it matches, we should
+	// prevent any further matches with absolute paths.
+	// If prompt path equals $W, set to '.'.
+	// If prompt path is a subdirectory of $HOME, then replace with '~'.
 	return `
 function prompter() {
 	local prompt_path
-	if [ -x "$(command -v realpath)" ]; then
-		prompt_path=$(realpath --relative-base=$W "$(pwd)")
-	else
-		prompt_path=$(basename "$(pwd)")
+
+	if [ -v W ]; then
+		prompt_path=${PWD/#$W\//}
```

Comment:
So if PWD were `$W/`, would prompt_path be `''` ? Is that an issue? If no, comment explaining that would help.

Reply:
PWD should never end with a `/`.